multisigverifier/src/lib.rs

```diff
@@ -16,29 +19,45 @@ pub trait Multisigverifier:
     #[endpoint(registerBridgeOps)]
     fn register_bridge_operations(
         &self,
-        hash_of_hashes: ManagedBuffer,
-        hash_of_bridge_ops: MultiValueEncoded<ManagedBuffer>,
+        bridge_operations_hash: ManagedBuffer,
         signature: BlsSignature<Self::Api>,
+        bridge_operations: MultiValueEncoded<TransferData<Self::Api>>
     ) {
         let caller = self.blockchain().get_caller();
-        let is_bls_valid = self.verify_bls(hash_of_bridge_ops.clone(), &signature, caller);
-        
-        if is_bls_valid { 
-            for hash in hash_of_bridge_ops {
-              self.pending_operations_mapper().insert(hash);
+        let is_bls_valid = self.verify_bls(&signature, caller, bridge_operations.clone());
+        let mut serialized_transferred_data = ManagedBuffer::new();
+
+        for operation in bridge_operations {
+            if let core::result::Result::Err(err) = operation.top_encode(&mut serialized_transferred_data) {
+                sc_panic!("Transfer data encode error: {}", err.message_bytes());
             }
         }
+        let transfers_data_sha256 = self.crypto().sha256(&serialized_transferred_data);
+        let transfers_data_hash = transfers_data_sha256.as_managed_buffer();
+        
+        require!(
+            bridge_operations_hash.eq(transfers_data_hash),
+            "Hash of all operations doesn't match the hash of transfer data"
+        );
+
+        require!(
+            is_bls_valid,
+            "BLS signature is not valid"
+        );
```

Comment:
Do this check right after the `verify_bls` call, no need to process further if that fails.

Reply:
done